controllers/predicate.go

```diff
@@ -0,0 +1,82 @@
+package controllers
+
+import (
+	pvcv1alpha1 "github.com/konveyor/volume-snapshot-mover/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+func datamoverBackupPredicate(scheme *runtime.Scheme) predicate.Predicate {
+	return predicate.Funcs{
+		// Update returns true if the Update event should be processed
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			if e.ObjectOld.GetGeneration() == e.ObjectNew.GetGeneration() {
+				return false
+			}
+			return isObjectOursBackup(scheme, e.ObjectOld)
+		},
+		// Create returns true if the Create event should be processed
+		CreateFunc: func(e event.CreateEvent) bool {
+			return isObjectOursBackup(scheme, e.Object)
+		},
+		// Delete returns true if the Delete event should be processed
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return !e.DeleteStateUnknown && isObjectOursBackup(scheme, e.Object)
+		},
+	}
+}
+
+// isObjectOurs returns true if the object is ours.
+// it first checks if the object has our group, version, and kind
+// else it will check for non empty OadpOperatorlabel labels
```

Comment:
NIT: `// else it will check for non empty VolumeSnapshotMoverController labels`

Reply:
good catch..copy paste error :D